contracts/Pool.sol

```diff
@@ -14,260 +15,368 @@ import "./IBorrowersRegistry.sol";
 
       __Ownable_init();
 
-      updateRates();
+      _updateRates();
     }
 
 
-    /* ========== SETTERS ========== */
+  /* ========== SETTERS ========== */
 
 
-    /**
-     * Sets the new rate calculator.
-     * The calculator is an external contract that contains the logic for calculating deposit and borrowing rates.
-     * Only the owner of the Contract can execute this function.
-     * @dev _ratesCalculator the address of rates calculator
-    **/
-    function setRatesCalculator(IRatesCalculator _ratesCalculator) external onlyOwner {
-        require(address(_ratesCalculator) != address(0), "The rates calculator cannot set to a null address");
+  /**
+   * Sets the new rate calculator.
+   * The calculator is an external contract that contains the logic for calculating deposit and borrowing rates.
+   * Only the owner of the Contract can execute this function.
+   * @dev _ratesCalculator the address of rates calculator
+  **/
+  function setRatesCalculator(IRatesCalculator ratesCalculator_) external onlyOwner {
+    require(address(ratesCalculator_) != address(0), "The rates calculator cannot set to a null address");
+    _ratesCalculator = ratesCalculator_;
+    _updateRates();
+  }
+
+
+  /**
+   * Sets the new borrowers registry contract.
+   * The borrowers registry decides if an account can borrow funds.
+   * Only the owner of the Contract can execute this function.
+   * @dev _borrowersRegistry the address of borrowers registry
+  **/
+  function setBorrowersRegistry(IBorrowersRegistry borrowersRegistry_) external onlyOwner {
+    require(address(borrowersRegistry_) != address(0), "The borrowers registry cannot set to a null address");
+
+    _borrowersRegistry = borrowersRegistry_;
+  }
 
-        ratesCalculator = _ratesCalculator;
-        updateRates();
-    }
 
+  /* ========== MUTATIVE FUNCTIONS ========== */
+  function transfer(address recipient, uint256 amount) external override returns (bool) {
+    require(msg.sender != address(0), "ERC20: transfer from the zero address");
+    require(recipient != address(0), "ERC20: transfer to the zero address");
+    require(recipient != address(this), "ERC20: cannot transfer to the pool address");
 
-    /**
-     * Sets the new borrowers registry contract.
-     * The borrowers registry decides if an account can borrow funds.
-     * Only the owner of the Contract can execute this function.
-     * @dev _borrowersRegistry the address of borrowers registry
-    **/
-    function setBorrowersRegistry(IBorrowersRegistry _borrowersRegistry) external onlyOwner {
-      require(address(_borrowersRegistry) != address(0), "The borrowers registry cannot set to a null address");
+    _accumulateDepositInterests(msg.sender);
 
-      borrowersRegistry = _borrowersRegistry;
+    require(_deposited[msg.sender] >= amount, "ERC20: transfer amount exceeds balance");
+
+    // (this is verified in "require" above)
+    unchecked {
```

Comment:
Even though it is being verified above, why do we still need to perform this step inside of an `unchecked` block?

Reply:
@jakub-wojciechowski

Reply:
Serves optimization since safety is checked before